### ISSUE TITLE
document all commands on the man pages

### ISSUE THREAD

**Issue by bart9h** [i3/i3#4211]
## I'm submitting a…

<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

I looked at the i3 manpages hoping for a reference on the `for_window` command, and found only a `(see HTML userguide)` message.

## Desired Behavior

The manpages should provide complete documentation.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.16.1 (2019-01-27) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16.1 (2019-01-27) (pid 2092)abort…)
Loaded i3 config: /home/doti/.config/i3/config (Last modified: Sun 27 Sep 2020 02:05:17 PM -03, 15 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
- Linux Distribution & Version: Devuan beowulf
- Are you using a compositor (e.g., xcompmgr or compton): No
</pre>

**Comment by orestisfl** [comment]
This is something I always wanted as well, if done automatically from the
current user guide and appended to the end of the man page.

**Comment by stapelberg** [comment]
Yeah, the manpage is lacking because nobody has time to maintain it.

I’m not sure how many people even look at the manpage. The HTML userguide is certainly where our effort is spent.

**Comment by orestisfl** [comment]
> I’m not sure how many people even look at the manpage. The HTML userguide is certainly where our effort is spent.

Well that's a chicken and egg problem, nobody is going to regularly look at the manpage if it is empty.

I think automatically appending the userguide so that it is accessible from `man i3` will be useful and more conforming to the usual behavior of nix tools. E.g. `man i3` and then `/for_window` is much more convenient than needing to open a browser tab with the userguide.

**Comment by stapelberg** [comment]
> Well that's a chicken and egg problem, nobody is going to regularly look at the manpage if it is empty.

It isn’t empty, it’s just not kept up-to-date as much as the userguide is.

I’m not sure we can say that it’s a chicken-and-egg problem. Given that we already have a manpage, I would expect to see more incremental contributions to improve said manpage if there was actual interest.

> I think automatically appending the userguide so that it is accessible from `man i3` will be useful and more conforming to the usual behavior of nix tools. E.g. `man i3` and then `/for_window` is much more convenient than needing to open a browser tab with the userguide.

Really depends on your workflow/environment. You and I use manpages and know how to navigate them, but I personally know a number of (advanced) computer users who never use manpages.

**Comment by orestisfl** [comment]
> I’m not sure we can say that it’s a chicken-and-egg problem. Given that we already have a manpage, I would expect to see more incremental contributions to improve said manpage if there was actual interest.

The main thing missing from the manpage is the userguide. I don't think we should accept a manual, incremental re-write of the userguide in the manpage.

> Really depends on your workflow/environment. You and I use manpages and know how to navigate them, but I personally know a number of (advanced) computer users who never use manpages.

So we should develop it for people like us that use it, people that don't will still have the userguide

**Comment by stapelberg** [comment]
> I don't think we should accept a manual, incremental re-write of the userguide in the manpage.

Agreed.

> So we should develop it for people like us that use it, people that don't will still have the userguide

No question about it, I just don’t have time to do it :)

**Comment by NorbertHauriel** [comment]
I was looking for an option flag in manual after finding it on stackoverflow while looking to see which config is used by i3:
**--more-version**, and these aliases also exist currently:
**-moreversion** (even the 3i issue template mentions this)
**-more-version**

Even if CLI docs are not updated often (thanks for the amazing html doc btw), I expect every program to include all help and version related commands at the very least.
To quote from the description of i3 man page:

> Please be aware that i3 is primarily targeted at advanced users and developers.

I think man page reading is a regular workflow for developers and system administrators to quickly recall some commands, especially useful debug commands like the above one.

I didn't want to create a new ticket for this, just my 2 cents. Can I help somehow?

**Comment by stapelberg** [comment]
The `--moreversion` flag is a special case altogether, as it isn’t even mentioned in the `--help` output. Maybe at this point we should just make `-v` (`--version`) trigger the `--moreversion` behavior and phase out `--moreversion`.
